tests/test_distributions.py

```diff
@@ -337,17 +330,6 @@ def test_contains() -> None:
     assert not ilu._contains(12.1)
     assert not ilu._contains(13)
 
-    # `step` is ignored and assumed to be 1.
-    iluq = distributions.IntLogUniformDistribution(low=2, high=7, step=2)
```

Comment:
`IntDistribution` doesn't allow `step` to be larger than `1`. The current `IntLogUniformDistribution` tolerates such step and replace it with `1` internally.

https://github.com/optuna/optuna/pull/3244/files#diff-2ff77057715ff5d7b689aa1ba37562d8a47052b826e0267493bfdd13ac7206e3L515-L522

This behavior is scheduled to be changed in v4.0.0 but we also stated that it is subject to change.
The impact of this change to users would be small and I think the current behavior is misleading.
I'd like to propose to change in this PR.

Reply:
:+1:

Reply:
👍